### DISCUSSION TITLE
Config API

### DISCUSSION THREAD

**Discussion by brandousov** [bluenviron/mediamtx#321] [General]
## Which version are you using?

v0.12.2

## Enhancement request

I think Server needs some HTTP/socket API for on-the-fly configuration changes with in-memory only or permanent `config.yml` modifications. For example look at: https://nginx.org/en/docs/http/ngx_http_upstream_conf_module.html.

It is very important for intencive work with creating/removing many streams under some hi-load REST API.

**Comment by ndaidong** [comment]
Great idea. I'm having to build a camera proxy service in which rtsp-simple-server is the core, with an API layer to receive the requests from outside, just for adding/removing the path items.

**Comment by aler9** [comment]
Yes, you're right

**Comment by dorinclisu** [comment]
I would be against this feature if it takes valuable time for @aler9 that could be spent improving the core instead.

Chances are the generic config API will not satisfy everyone's needs (like oauth2 authentication, in my case) so it will have to be wrapped by another application API anyway, thereby reducing the benefit.

Keep it simple, it's very easy to make an API in python using flask / fastapi that writes to file using pyaml.

**Comment by aler9** [comment]
an HTTP API is available in v0.17.0

[a comment by an automated account is not shown]
